### DISCUSSION TITLE
Using backtick key for hyersnips expansion

### DISCUSSION THREAD

**Discussion by TryerGit** [sleepymalc/VSCode-LaTeX-Inkscape#19] [Q&A] [answered]
Hello,

I have found hypersnips faster than native LaTeX Workshop snippets for typing greek symbols. I would like to use the backtick ` key followed by a mnemonic for some snippets such as:

```
context math(context)
snippet `a "alpha" iA
\alpha 
endsnippet
```

At present, in LaTeX Workshop, the backtick key gets autocompleted due to the following setting:

        "autoClosingPairs": [
             ["`", "'"]
        ]

in latex-language-configuration.json. See [here](https://github.com/James-Yu/LaTeX-Workshop/blob/f8a8257f3fcbc02b4e26c255a68bb98b46f6a9f6/syntax/latex-language-configuration.json#L74-L101)

As a result, when I press backtick, I end up with `|' where | is the cursor position. 

Pressing `a` after this expands to:

`\alpha '`

After trigger expansion, is it possible to automatically delete the trailing `'` so that one is left with `\alpha ` ?

Thank you.

**Comment by sleepymalc** [comment, marked as answer]
If you're sure that this auto closing will only be triggered in normal scope (non-math), you can implement it in Hypersnips and define a new scope function called normal which return the negation of `math()`, and use it for your new snippet defined in Hypersnips as 
```javascript
context normal(context)
snippet ` "auto-closing" iA
`$1' $0
endsnippet
```

**Reply by TryerGit** [reply]
Let me think about it. I really want to use backtick as snippet triggers because, amongst other things, it is a single keypress as opposed to LW's default `@` which is two keystrokes and it is also consistent with VimTeX in insert mode for greek symbols.

Currently, my workaround is somewhat hackish -- alter the code of LW by doing [this](https://github.com/James-Yu/LaTeX-Workshop/discussions/3623#discussioncomment-4678009). Let me think through your suggestion and see if it is better.

Thanks!

**Reply by sleepymalc** [reply]
for me, I use a combination of `;` and simply typing out the name of the greek symbols. For example, I usually use `;a` for $\alpha$, while using `pi` for $\pi$ since `;p` is not fast enough (don't use your pinky twice!) I think this is faster since the backtick is far from the main row on your keyboard. Might just give up on this idea once and for all.
